docs/Coding-Guidelines/WinRT-Registration.md

```diff
@@ -0,0 +1,144 @@
+# WinRT Registration
+
+Windows App SDK provides registers WinRT types via *manifested* solutions:
+
+* WinAppSDK used via MSIX packages declare types in their `appxmanifest.xml`
+* WinAppSDK used via Self-Contained declare types in a SxS (aka Fusion) manifest
+  (`application.manifest`) that must be compiled into the caller's PE file
+
+## Registration Options
+
+Before discussing further it helps to understand the options available and their tradeoffs.
+
+There are 3 ways to resolve a packaged WinRT's activatable class id (ACID):
+
+A. **Via the process' package graph** -- Requires using WinAppSDK/MSIX and the ACID defined in
+WinAppSDK's `appxmanifest.xml`
+
+B. **Via RegFreeWinRT** -- Requires the ACID defined in the caller's SxS (Fusion) manifest and the
+DLL found via LoadLibrary
+
+C. **Via language projections** -- C#/WinRT and C++/WinRT support the same "we'll resolve it
+ourselves" option crawling the filesystem looking for a DLL containing the implementation. Requires
+caller uses a language project (C#/WinRT, C++/WinRT, etc), the WinRT runtimeclass implementation is
+in a DLL and the DLL's filename matches the ACID's namespace/prefix.
+
+WinAppSDK/MSIX uses option A.
+
+WinAppSDK/Self-Contained uses option B.
+
+## Why use manifested registration?
+
+WinAppSDK uses options A+B ByDesign, for performance and complexity reasons.
```

Comment:
Just state the two names here, "WinAppSDK uses both the process' package graph and registration-free WinRT." Then go on to explain the precise reasoning if you see fit. No need to tease the "for perf and cplx" parts.